src/util/ConfigManager/ConfigManager.h

```diff
@@ -114,7 +114,8 @@ class ConfigManager {
     @param instance The `HashMapEntry` you want this from.
     */
     template <typename ReturnType>
-    requires isTypeContainedIn<ReturnType, Data> && std::is_object_v<ReturnType>
+    requires SimilarToAnyTypeIn<ReturnType, Data> &&
+             std::is_object_v<ReturnType>
```

Comment:
In your whole benchmark module you have some concepts that appear multiple times
e.g. `any of the AvailableTypes`  or `ConfigOptionOrManager`. If you assign these separate names, then the code gets shorter,more readable, and maintainable.